src/crypto/mbedtls/verifier.cpp

```diff
@@ -108,52 +108,68 @@ namespace crypto
   class CertificateChain
   {
   public:
-    size_t n = 0;
-    mbedtls_x509_crt* raw = NULL;
+    mbedtls_x509_crt raw;
 
-    CertificateChain(const std::vector<const Pem*>& certs)
+    CertificateChain()
     {
-      if (!certs.empty())
-      {
-        n = certs.size();
-        raw = new mbedtls_x509_crt[certs.size()];
-
-        for (size_t i = 0; i < certs.size(); i++)
-        {
-          mbedtls_x509_crt_init(&raw[i]);
-        }
+      mbedtls_x509_crt_init(&raw);
+    }
 
-        for (size_t i = 0; i < certs.size(); i++)
+    void add(const std::vector<const Pem*>& certs)
+    {
+      for (size_t i = 0; i < certs.size(); i++)
```

Comment:
for(auto & cert: certs) perhaps, since you don't seem to use the index? Alternatively, include the index in the exception message (could not parse PEM in chain at position {i}...)?